lib/puppet/pops/loaders.rb

```diff
@@ -63,6 +63,11 @@ def initialize(environment, for_agent)
     # 4. module loaders are set up from the create_environment_loader, they register themselves
   end
 
+  # Called after loader has been added to Puppet Context as :loaders so that dynamic types can
+  # be pre-loaded with a fully configured loader system
+  def pre_load
```

Comment:
Is `pre_load` a good name? isn't this more of a "on_load_config_complete" or similar, on which some loaders may do loading of elements?
Alternatively, this is some kind of "before" call (with a guarantee that it happens before some other type of event) - for example "before_evaluation" or "before_dynamic_loading" - or is it perhaps "finish_configuration" ? (Feel free to stick with "pre_load" if you disagree).